### PR TITLE
Add Grafana data source configuration instruction for Quickwit Searcher Lambda

### PR DESCRIPTION
**Pull request by tyrwzl**
### Description

add instructions to set up Grafana for Quickwit Searcher Lambda.
related PR: https://github.com/quickwit-oss/quickwit/pull/5069#issuecomment-2147449057

### How was this PR tested?

N/A

### CODE REVIEW

File: distribution/lambda/README.md [hunk 1]
```diff
@@ -139,3 +139,24 @@ curl -d '{"query":"quantity:>5", "max_hits": 10}' -H "Content-Type: application/
...
  * `cdk diff`        compare deployed stack with current state
  * `cdk docs`        open CDK documentation
+
+### Grafana data source setup
+
+You can query and visualize Quickwit Searcher Lambda data from Grafana by using Quickwit data source for Grafana.
+
+#### Prerequisites
+
+- [Setup HTTP API endpoint for Quickwit Searcher Lambda](#set-up-a-search-api)
```
Comment (guilload) [inline distribution/lambda/README.md:149]:
```suggestion
- [Set up HTTP API endpoint for Quickwit Searcher Lambda](#set-up-a-search-api)
```

File: distribution/lambda/README.md [hunk 2]
```diff
@@ -139,3 +139,24 @@ curl -d '{"query":"quantity:>5", "max_hits": 10}' -H "Content-Type: application/
...
+
+You need to provide the following information.
+
+|Variable|Description|Example|
+|--|--|--|
+|HTTP URL| HTTP search endpoint for Quickwit Searcher Lambda | https://*******.execute-api.us-east-1.amazonaws.com/api/v1 |
+|Custom HTTP Headers| (Optional) If you configure API Gateway to require an API key, set `x-api-key` HTTP Header | Header: `x-api-key` <br> Value: API key value|
+|Index ID| Same as `QW_LAMBDA_INDEX_ID` | hdfs-logs |
+
+After enter the above values, click "Save & test" and you can query Quickwit data from Grafana!
```
Comment (guilload) [inline distribution/lambda/README.md:162]:
```suggestion
After entering the above values, click "Save & test" and you can query Quickwit data from Grafana!
```

**Review by guilload — COMMENTED** [review]
Left some minor comments. I'll let @rdettai take a look and approve.

File: distribution/lambda/README.md [hunk 3]
```diff
@@ -139,3 +139,24 @@ curl -d '{"query":"quantity:>5", "max_hits": 10}' -H "Content-Type: application/
  * `cdk deploy`      deploy this stack to your default AWS account/region
  * `cdk diff`        compare deployed stack with current state
  * `cdk docs`        open CDK documentation
+
+### Grafana data source setup
+
+You can query and visualize Quickwit Searcher Lambda data from Grafana by using Quickwit data source for Grafana.
```
Comment (rdettai) [inline distribution/lambda/README.md:145]:
```suggestion
You can query and visualize the Quickwit Searcher Lambda from Grafana by using the [Quickwit data source for Grafana](https://grafana.com/grafana/plugins/quickwit-quickwit-datasource/).
```

File: distribution/lambda/README.md [hunk 4]
```diff
@@ -110,7 +110,7 @@ Searcher only:
 | QW_LAMBDA_PARTIAL_REQUEST_CACHE_CAPACITY | `searcher.partial_request_cache_capacity` node config | 64M |
 
 
-### Set up a search API
+### <a name=set-up-a-search-api />Set up a search API
```
Comment (rdettai) [inline distribution/lambda/README.md:113]:
```suggestion
### Set up a search API
```

you can ref the title as `[link](set-up-a-search-api)` without the `<a/>` tag

File: distribution/lambda/README.md [hunk 5]
```diff
@@ -139,3 +139,24 @@ curl -d '{"query":"quantity:>5", "max_hits": 10}' -H "Content-Type: application/
...
+- [Install Quickwit data source plugin on Grafana](https://github.com/quickwit-oss/quickwit-datasource#installation)
+
+#### Configure Grafana data source
+
+You need to provide the following information.
+
+|Variable|Description|Example|
+|--|--|--|
+|HTTP URL| HTTP search endpoint for Quickwit Searcher Lambda | https://*******.execute-api.us-east-1.amazonaws.com/api/v1 |
+|Custom HTTP Headers| (Optional) If you configure API Gateway to require an API key, set `x-api-key` HTTP Header | Header: `x-api-key` <br> Value: API key value|
```
Comment (rdettai) [inline distribution/lambda/README.md:159]:
```suggestion
|Custom HTTP Headers| If you configure API Gateway to require an API key, set `x-api-key` HTTP Header | Header: `x-api-key` <br> Value: API key value|
```

File: distribution/lambda/README.md [hunk 6]
```diff
@@ -139,3 +139,24 @@ curl -d '{"query":"quantity:>5", "max_hits": 10}' -H "Content-Type: application/
...
+|Index ID| Same as `QW_LAMBDA_INDEX_ID` | hdfs-logs |
+
+After entering the above values, click "Save & test" and you can query Quickwit data from Grafana!
```
Comment (rdettai) [inline distribution/lambda/README.md:162]:
```suggestion
After entering these values, click "Save & test" and you can now query your Quickwit Lambda from Grafana!
```

**Review by rdettai — APPROVED** [review]

**Pull request merged**